lib/filter-gmail-compiler.js

```diff
@@ -0,0 +1,131 @@
+// Compiler that transforms filter expression AST into a gmail search expression.
+//
+// The expression is returned from the compile method.
+
+var ASTVisitor = require('juttle/lib/compiler/ast-visitor');
+var JuttleErrors = require('juttle/lib/errors');
+var JuttleMoment = require('juttle/lib/moment').JuttleMoment;
+var _ = require('underscore');
+
+// FilterGmailCompiler derives from ASTVisitor which provides a way to
+// traverse the abstract syntax tree that the juttle compiler
+// generates for the read command's filter expression.
+//
+// While traversing the tree, callbacks are called for the various
+// parts of the filter expression. The FilterGmailCompiler object maps
+// individual items in the tree into appropriate gmail advanced search
+// (https://support.google.com/mail/answer/7190?hl=en) terms. As the
+// tree is traversed, these items are combined into a complete gmail
+// search expression. This is used when reading messages.
+
+var FilterGmailCompiler = ASTVisitor.extend({
+    initialize: function(options) {
+        this.location = options.location;
+        this.source = options.source;
+        this.supported_headers = ["from", "to", "subject", "cc", "bcc"];
```

Comment:
Nit: This could be moved out into a "const" above instead of being an instance variable.